OpenCVConfig.cmake is broken on windows

If I build program with cmake, it can not find opencv.

```
-- OpenCV ARCH: x64
-- OpenCV RUNTIME: vc14
-- OpenCV STATIC: OFF
CMake Warning at D:/env/conda64_3.7/Library/OpenCVConfig.cmake:166 (message):
  Found OpenCV Windows Pack but it has no binaries compatible with your
  configuration.

  You should manually point CMake variable OpenCV_DIR to your build of OpenCV
  library.
Call Stack (most recent call first):
  CMakeLists.txt:85 (FIND_PACKAGE)


CMake Error at CMakeLists.txt:85 (FIND_PACKAGE):
  Found package configuration file:

    D:/env/conda64_3.7/Library/OpenCVConfig.cmake

  but it set OpenCV_FOUND to FALSE so package "OpenCV" is considered to be
  NOT FOUND.


-- Configuring incomplete, errors occurred!
```

``` cmake
# in OpenCVConfig.cmake
if(OpenCV_RUNTIME AND OpenCV_ARCH)
  if(OpenCV_STATIC AND EXISTS "${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib/OpenCVConfig.cmake")
    if(OpenCV_CUDA AND EXISTS "${OpenCV_CONFIG_PATH}/gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib/OpenCVConfig.cmake")
      set(OpenCV_LIB_PATH "${OpenCV_CONFIG_PATH}/gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib")
    else()
      set(OpenCV_LIB_PATH "${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib")
    endif()
  elseif(EXISTS "${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib/OpenCVConfig.cmake")
    if(OpenCV_CUDA AND EXISTS "${OpenCV_CONFIG_PATH}/gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib/OpenCVConfig.cmake")
      set(OpenCV_LIB_PATH "${OpenCV_CONFIG_PATH}/gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib")
    else()
      set(OpenCV_LIB_PATH "${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib")
    endif()
  endif()
endif()
```

In conda, the lib folder is not located at ${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib.

same problem occurring here:

```
E:\Desktop\opencv-test\build> cmake -G "MinGW Makefiles" ..
-- The C compiler identification is GNU 7.2.0
-- The CXX compiler identification is GNU 7.2.0
-- Check for working C compiler: C:/mingw64/i686-7.2.0-posix-dwarf-rt_v5-rev0/bin/gcc.exe
-- Check for working C compiler: C:/mingw64/i686-7.2.0-posix-dwarf-rt_v5-rev0/bin/gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/mingw64/i686-7.2.0-posix-dwarf-rt_v5-rev0/bin/g++.exe
-- Check for working CXX compiler: C:/mingw64/i686-7.2.0-posix-dwarf-rt_v5-rev0/bin/g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- OpenCV ARCH: x86
-- OpenCV RUNTIME: mingw
-- OpenCV STATIC: ON
CMake Warning at C:/Miniconda3x64/pkgs/opencv-3.4.1-py36_200/Library/OpenCVConfig.cmake:140 (message):
  Found OpenCV Windows Pack but it has no binaries compatible with your
  configuration.

  You should manually point CMake variable OpenCV_DIR to your build of OpenCV
  library.
Call Stack (most recent call first):
  CMakeLists.txt:4 (find_package)


CMake Error at CMakeLists.txt:4 (find_package):
  Found package configuration file:

    C:/Miniconda3x64/pkgs/opencv-3.4.1-py36_200/Library/OpenCVConfig.cmake

  but it set OpenCV_FOUND to FALSE so package "OpenCV" is considered to be
  NOT FOUND.


-- Configuring incomplete, errors occurred!
See also "E:/Desktop/opencv-test/build/CMakeFiles/CMakeOutput.log".
```

and 

```
E:\Desktop\opencv-test\build> conda info

     active environment : None
       user config file : C:\Users\carlo\.condarc
 populated config files : C:\Users\carlo\.condarc
          conda version : 4.5.1
    conda-build version : 3.8.0
         python version : 3.6.5.final.0
       base environment : C:\Miniconda3x64  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/win-64
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Miniconda3x64\pkgs
                          C:\Users\carlo\AppData\Local\conda\conda\pkgs
       envs directories : C:\Miniconda3x64\envs
                          C:\Users\carlo\AppData\Local\conda\conda\envs
                          C:\Users\carlo\.conda\envs
               platform : win-64
             user-agent : conda/4.5.1 requests/2.18.4 CPython/3.6.5 Windows/10 Windows/10.0.16299
          administrator : False
             netrc file : None
           offline mode : False
```

--- a/recipe/CMakeLists.txt
+++ b/recipe/CMakeLists.txt
@@ -1,0 +1,13 @@
+# Mostly follows <opencv>/samples/cpp/example_cmake/CMakeLists.txt
+cmake_minimum_required(3.5.1)
+project(cmake_build_test)
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+find_package(OpenCV REQUIRED)
+message(STATUS "OpenCV library status:")
+message(STATUS "    config: ${OpenCV_DIR}")
+message(STATUS "    version: ${OpenCV_VERSION}")
+message(STATUS "    libraries: ${OpenCV_LIBS}")
+message(STATUS "    include path: ${OpenCV_INCLUDE_DIRS}")
+add_executable(cmake_build_test test.cpp)
+target_link_libraries(cmake_build_test LINK_PRIVATE ${OpenCV_LIBS})

--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -32,6 +32,10 @@ cmake -LAH -G "Ninja"                                                           
     -DCMAKE_BUILD_TYPE="Release"                                                    ^
     -DCMAKE_INSTALL_PREFIX=%UNIX_LIBRARY_PREFIX%                                    ^
     -DCMAKE_PREFIX_PATH=%UNIX_LIBRARY_PREFIX%                                       ^
+    -DOPENCV_CONFIG_INSTALL_PATH=cmake                                              ^
+    -DOPENCV_BIN_INSTALL_PATH=bin                                                   ^
+    -DOPENCV_LIB_INSTALL_PATH=lib                                                   ^
+    -DOPENCV_GENERATE_SETUPVARS=OFF                                                 ^
     -DOPENCV_DOWNLOAD_TRIES=1;2;3;4;5                                               ^
     -DOPENCV_DOWNLOAD_PARAMS=INACTIVITY_TIMEOUT;30;TIMEOUT;180;SHOW_PROGRESS        ^
     -DWITH_LAPACK=1                                                                 ^
@@ -64,8 +68,6 @@ cmake -LAH -G "Ninja"                                                           
     -DWITH_QT=5                                                                     ^
     -DINSTALL_C_EXAMPLES=0                                                          ^
     -DOPENCV_EXTRA_MODULES_PATH=%UNIX_SRC_DIR%/opencv_contrib/modules               ^
-    -DEXECUTABLE_OUTPUT_PATH=%UNIX_LIBRARY_BIN%                                     ^
-    -DLIBRARY_OUTPUT_PATH=%UNIX_LIBRARY_BIN%                                        ^
     -DPYTHON_EXECUTABLE=""                                                          ^
     -DPYTHON_INCLUDE_DIR=""                                                         ^
     -DPYTHON_PACKAGES_PATH=""                                                       ^
@@ -98,16 +100,4 @@ if errorlevel 1 exit 1
 cmake --build . --target install --config Release
 if errorlevel 1 exit 1
 
-if "%ARCH%" == "32" ( set "OPENCV_ARCH=86")
-if "%ARCH%" == "64" ( set "OPENCV_ARCH=64")
-
-robocopy %LIBRARY_PREFIX%\x%OPENCV_ARCH%\vc%VS_MAJOR%\ %LIBRARY_PREFIX%\ *.* /E
-if %ERRORLEVEL% GEQ 8 exit 1
-
-rem Remove files installed in the wrong locations
-rd /S /Q "%LIBRARY_BIN%\Release"
-rd /S /Q "%LIBRARY_PREFIX%\x%OPENCV_ARCH%"
-rem RD is a bit horrible and doesn't return an errorcode properly, so
-rem the errorcode from robocopy is propagated (which is non-zero), so we
-rem forcibly exit 0 here
 exit 0

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -145,7 +145,7 @@ test:
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
         - mkdir -p cmake_build_test && pushd cmake_build_test
-        - cmake -G "Ninja" -D "OpenCV_DIR=$PREFIX/cmake" "$RECIPE_DIR"
+        - cmake -G "Ninja" -D "OpenCV_DIR={{ PREFIX }}/cmake" "${{ RECIPE_DIR }}"
         - cmake --build . --config Release
         - popd
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -144,7 +144,7 @@ test:
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
         - mkdir -p cmake_build_test && pushd cmake_build_test
-        - cmake -G "${CMAKE_GENERATOR}" -D "OpenCV_DIR=${PREFIX}/cmake" "${RECIPE_DIR}"
+        - cmake -G "$CMAKE_GENERATOR" -D "OpenCV_DIR=$PREFIX/cmake" "$RECIPE_DIR"
         - cmake --build . --config Release
         - popd
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -145,7 +145,7 @@ test:
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
         - mkdir -p cmake_build_test && pushd cmake_build_test
-        - cmake -G "Ninja" -D "OpenCV_DIR={{ PREFIX }}/cmake" "${{ RECIPE_DIR }}"
+        - cmake -G "Ninja" -D "OpenCV_DIR={{ PREFIX }}/cmake" "{{ RECIPE_DIR }}"
         - cmake --build . --config Release
         - popd
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -79,6 +79,7 @@ test:
       - liblapack * *openblas         # [py==36]
       - liblapack * *mkl              # [py==37]
       - cmake
+      - ninja
     imports:
       - cv2
       - cv2.xfeatures2d
@@ -144,7 +145,7 @@ test:
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
         - mkdir -p cmake_build_test && pushd cmake_build_test
-        - cmake -G "$CMAKE_GENERATOR" -D "OpenCV_DIR=$PREFIX/cmake" "$RECIPE_DIR"
+        - cmake -G "Ninja" -D "OpenCV_DIR=$PREFIX/cmake" "$RECIPE_DIR"
         - cmake --build . --config Release
         - popd
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -19,7 +19,7 @@ source:
     folder: opencv_contrib
 
 build:
-  number: 5
+  number: 6
   # Python2.7 support dropped: https://github.com/opencv/opencv/issues/8481
   skip: true  # [win and py27]
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -147,7 +147,7 @@ test:
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
         - mkdir -p cmake_build_test && pushd cmake_build_test
-        - cmake -G "Ninja" -D "OpenCV_DIR={{ PREFIX }}/cmake" ..
+        - cmake -G "Ninja" ..
         - cmake --build . --config Release
         - popd
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -83,7 +83,8 @@ test:
     imports:
       - cv2
       - cv2.xfeatures2d
-
+    files:
+      - CMakeLists.txt
     commands:
         # Verify dynamic libraries on all systems
         {% set win_ver_lib = version|replace(".", "") %}
@@ -145,7 +146,7 @@ test:
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
         - mkdir -p cmake_build_test && pushd cmake_build_test
-        - cmake -G "Ninja" -D "OpenCV_DIR={{ PREFIX }}/cmake" "{{ RECIPE_DIR }}"
+        - cmake -G "Ninja" -D "OpenCV_DIR={{ PREFIX }}/cmake" ..
         - cmake --build . --config Release
         - popd
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -78,6 +78,7 @@ test:
       # One test done on different versions of python on each platform
       - liblapack * *openblas         # [py==36]
       - liblapack * *mkl              # [py==37]
+      - cmake
     imports:
       - cv2
       - cv2.xfeatures2d
@@ -142,6 +143,10 @@ test:
         - test -f $PREFIX/lib/libopencv_freetype${SHLIB_EXT}     # [unix]
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
+        - mkdir -p cmake_build_test && pushd cmake_build_test
+        - cmake -G "${CMAKE_GENERATOR}" -D "OpenCV_DIR=${PREFIX}/cmake" "${RECIPE_DIR}"
+        - cmake --build . --config Release
+        - popd
 
 about:
   home: http://opencv.org/

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -85,6 +85,7 @@ test:
       - cv2.xfeatures2d
     files:
       - CMakeLists.txt
+      - test.cpp
     commands:
         # Verify dynamic libraries on all systems
         {% set win_ver_lib = version|replace(".", "") %}

